### ISSUE TITLE
Comparison between chainer-Waveglow & Nvidia official implementation

### ISSUE THREAD

**Issue by deepconsc** [dhgrs/chainer-WaveGlow#3]
I'm interested, what is the generation time in case of this implementation, and which mel generator is the model optimized on. 

Additionally, trying to understand the difference, or improvement besides the probability density into probability mass convertation.

**Comment by dhgrs** [comment]
- Generation time
I used [this profiler](https://gist.github.com/dhgrs/7bac137a70442b3fd790cfb8f7d442ab). It's about x30 faster than realtime on 1080Ti.

- The difference
The loss function is same as NVIDIA's official implementation or paper. Probability mass is not used to train. It is used for only logging. And how to convert probability density into probability mass is written in [this comment](https://github.com/openai/glow/issues/43#issuecomment-414142236).

**Comment by deepconsc** [comment]
Gotcha! 
Additional question - can you describe the profiler & its function in couple of words? 
In other words - what it does. 

And, second one - x300 faster than realtime is a huge, I mean huge improvement. Is the quality the same as provided in your samples on soundcloud? (https://soundcloud.com/dhgrs/sets/waveglow).

And, how language indepedent is the model, if I use it to fine-tune on different language dataset, and generate mels using Tacotron2.

Thnx.

**Comment by dhgrs** [comment]
This is output of `profiler.py`.
```
chainer-WaveGlow# python3 profiler.py -g 0 -l 1
Generate 0.9984580498866213 secs...
It took 0.04848980903625488 secs to generate.
It took 0.0385404109954834 secs to generate.
It took 0.035938572883605954 secs to generate.
It took 0.03309469223022461 secs to generate.
It took 0.03186588287353516 secs to generate.
It took 0.03151259422302246 secs to generate.
It took 0.030570030212402344 secs to generate.
It took 0.031554555892944335 secs to generate.
It took 0.031134033203125 secs to generate.
It took 0.031122517585754395 secs to generate.
```
And I'm very sorry that x300 is mistake. x30 faster is correct. I fixed my old comment.

Last, I only tried to generate from only LJ-Speech's mels. So I don't know how it works on other language dataset or generated mels. Please report it if you try it.

**Comment by deepconsc** [comment]
I'm trying to run it on different dataset rn. I'll use the pre-trained model on 8 v100 gpus for around 100k iters, and let's see the result. I'll update you as soon as it's done
